File: qosten-react/check_explanations.py
import json

input_file = r'C:\Users\DFIT\Downloads\question-bank-full-2026-06-04.json'

try:
    with open(input_file, 'r', encoding='utf-8') as f:
        data = json.load(f)

    bangla_2nd = [item for item in data if item.get('subject') == 'বাংলা দ্বিতীয় পত্র']
    total = len(bangla_2nd)
    filled = sum(1 for item in bangla_2nd if item.get('explanation') and item.get('explanation').strip())
    empty = total - filled

    print(f'Total "বাংলা দ্বিতীয় পত্র" questions: {total}')
    print(f'Filled explanations: {filled}')
    print(f'Empty explanations: {empty}')
    if total > 0:
        print(f'Fill rate: {(filled/total)*100:.2f}%')
except Exception as e:
    print(f'Error: {e}')

File: qosten-react/check_explanations.mjs
import fs from "fs";

const inputFile = "C:\\Users\\DFIT\\Downloads\\question-bank-full-2026-06-04.json";

try {
  const data = JSON.parse(fs.readFileSync(inputFile, "utf8"));
  const bangla2nd = data.filter(item => item.subject === "বাংলা দ্বিতীয় পত্র");
  
  const total = bangla2nd.length;
  const filled = bangla2nd.filter(item => item.explanation && item.explanation.trim()).length;
  const empty = total - filled;

  console.log(`Total "বাংলা দ্বিতীয় পত্র" questions: ${total}`);
  console.log(`Filled explanations: ${filled}`);
  console.log(`Empty explanations: ${empty}`);
  if (total > 0) {
    console.log(`Fill rate: ${((filled / total) * 100).toFixed(2)}%`);
  }
} catch (err) {
  console.error("Error:", err.message);
}
